src/qtt/instrument_drivers/virtualAwg/virtual_awg.py

```diff
@@ -87,6 +91,28 @@ def __set_parameters(self):
             self.add_parameter('digitizer_marker_delay', initial_value=0, set_cmd=None, unit='s')
             self.add_parameter('digitizer_marker_uptime', initial_value=1e-8, set_cmd=None, unit='s')
 
+    @property
+    def settings(self) -> Instrument:
+        return self._settings
+
+    @settings.setter
+    def settings(self, value: Instrument) -> None:
```

Comment:
This method can only be called once, since `__set_parameters` (which used `add_parameters) will throw an error if called twice.
It is already called in the constructor, so that makes it not suitable as a public interface. Updating the settings is something we do in the lab sometimes.

Reply:
This method can only be called once indeed. Reworked it now such that it can be done after construction (this is required by the ticket)